packages/aws-cdk/lib/api/util/placeholders.ts

```diff
@@ -20,3 +20,21 @@ export async function replaceEnvPlaceholders<A extends { }>(object: A, env: cxap
   });
 }
 
+/**
+ * Replace the {ACCOUNT} and {REGION} placeholders in all strings found in a complex object.
+ */
+export async function replaceEnvPlaceholdersv3<A extends { }>(object: A, env: cxapi.Environment, sdkProvider: SdkProviderv3): Promise<A> {
```

Comment:
Instead of duplicating this function, another way of doing this is:

1. Create a new interface that only has the `baseCredentialsPartition` method.
2. Both `SdkProvider` and `SdkProviderv3` implement it.
3. Change the last parameter of this method to be of the new interface.

This doesn't change any behavior and we get to write less code. No need for `isSdkProviderV3` and its usage, for example.

Reply:
I think I need to change this code anyway but I need to figure that out today, TBD on this suggestion until I figure out what I'm doing with `provideContextValues`